src/Extension/AccountReset/SecurityExtension.php

```diff
@@ -0,0 +1,160 @@
+<?php declare(strict_types=1);
+
+namespace SilverStripe\MFA\Extension\AccountReset;
+
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+use SilverStripe\Core\Extensible;
+use SilverStripe\Core\Extension;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\Form;
+use SilverStripe\Forms\FormAction;
+use SilverStripe\Forms\PasswordField;
+use SilverStripe\Forms\RequiredFields;
+use SilverStripe\MFA\JSONResponse;
+use SilverStripe\MFA\RequestHandler\BaseHandlerTrait;
+use SilverStripe\ORM\FieldType\DBDatetime;
+use SilverStripe\ORM\ValidationResult;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\Security;
+
+/**
+ * Extends the Security controller to support Account Resets. This extension can
+ * itself be extended to add procedures to the reset action (such as removing
+ * additional authentication factors, sending alerts, etc.)
+ *
+ * @package SilverStripe\MFA\Extension
+ * @property Security owner
+ */
+class SecurityExtension extends Extension
+{
+    use JSONResponse, BaseHandlerTrait, Extensible;
+
+    private static $url_handlers = [
+        'GET reset-account' => 'resetaccount',
+        'ResetAccountForm' => 'ResetAccountForm',
+    ];
+
+    private static $allowed_actions = [
+        'resetaccount',
+        'ResetAccountForm',
+    ];
+
+    public function resetaccount(HTTPRequest $request): HTTPResponse
+    {
+        if (Security::getCurrentUser()) {
+            return $this->jsonResponse(['error' => 'Already authenticated'], 400);
+        }
+
+        ['m' => $memberID, 't' => $token] = $request->getVars();
+
+        /** @var Member|MemberExtension $member */
+        $member = Member::get()->byID($memberID);
+
+        if (is_null($member) || $member->verifyAccountResetToken($token) === false) {
+            return $this->jsonResponse(['error' => 'Invalid member or token'], 400);
+        }
```

Comment:
Do you need to enforce sudo mode for this action?

Reply:
I don't think sudo mode _can_ be enforced during an account reset - you don't have a password to provide whilst you're resetting your password.